cdap-ui/app/cdap/components/DataPrep/Directives/Calculate/index.js

```diff
@@ -0,0 +1,616 @@
+      inputLabel = T.translate(`${PREFIX}.destinationColumnInputLabel`);
+      inputPlaceholder = T.translate(`${PREFIX}.destinationColumnInputPlaceholder`);
+    }
+
+    return (
+      <div>
+        <div className={classnames({"bigger-new-column-label": this.state.operationPopoverOpen === 'CHARCOUNT'})}>
+          {inputLabel}
+        </div>
+        <Input
+          type="text"
+          className="form-control"
+          value={this.state.newColumnInput}
+          onChange={this.setNewColumnInput}
+          onKeyPress={this.handleKeyPress}
+          placeholder={inputPlaceholder}
+          autoFocus
+        />
+        {
+          columnNameAlreadyExists(this.state.newColumnInput) ? (
+            <WarningContainer
+              message={T.translate(`${PREFIX}.newColumnInputDuplicate`)}
+            />
+          ) : null
+        }
+      </div>
+    );
+  }
+
+  renderActionButtons() {
+    return (
+      <div className="action-buttons">
+        <button
+          className="btn btn-primary float-xs-left"
+          disabled={(this.state.createNewColumn || this.state.operationPopoverOpen === 'CHARCOUNT') && this.state.newColumnInput.length === 0}
+          onClick={this.getExpressionAndApply}
+        >
+          {T.translate('features.DataPrep.Directives.apply')}
+        </button>
+
+        <button
+          className="btn btn-link float-xs-right"
+          onClick={this.setDefaultState}
+        >
+          {T.translate('features.DataPrep.Directives.cancel')}
+        </button>
+      </div>
+    );
+  }
+
+  renderThirdLevelPopover() {
+    if (!this.state.operationPopoverOpen) { return null; }
+
+    if (this.state.operationPopoverOpen === 'CHARCOUNT') {
+      return (
+        <div
+          className="third-level-popover"
+          onClick={preventPropagation}
+        >
+          {this.renderNewColumnNameInput()}
+
+          <hr />
+
+          {this.renderActionButtons()}
+        </div>
+      );
+    }
+
+    let operationSign;
+
+    switch (this.state.operationPopoverOpen) {
+      case 'ADD':
+        operationSign = '+';
+        break;
+      case 'SUBTRACT':
+        operationSign = '-';
+        break;
+      case 'MULTIPLY':
+        operationSign = 'x';
+        break;
+      case 'DIVIDE':
+        operationSign = '/';
+        break;
+      case 'MODULO':
+        operationSign = '%';
+        break;
+    }
+
+    return (
+      <div
+        className="third-level-popover"
+        onClick={preventPropagation}
+      >
+        {
+          this.state.operationPopoverOpen === 'POWEROF' ?
+            (
+              <div className="operation-description">
+                {T.translate(`${PREFIX}.Options.POWEROF.description`)}
+              </div>
+            )
+          : null
+        }
+        <div className="operation-sign-input">
+          <span className="operation-sign">
+            {operationSign}
+          </span>
+          <Input
+            className="operation-input"
```

Comment:
I'm not sure if we should disable it, since negative values are valid operands in these operations, except modulo. I can make the minimum input value to be 1 for modulo